src/modules.h

```diff
@@ -134,6 +134,62 @@ struct init_info {
 };
 
 
+/****************************************************************************
+**
+**  Some helper functions and macros used for validation in GVAR_FUNC_2ARGS
+**  and its likes.
+**
+**  The trick is that VALIDATE_FUNC_nARGS(func) produces code that the
+**  compiler can trivially prove to be equivalent to just inserting 'func';
+**  and the "call" to VALIDATE_FUNC_HELPER_n can never ever be executed;
+**  but since it is still there in the input, the compiler has to check
+**  that its argument has the correct type.
+*/
+static inline ObjFunc VALIDATE_FUNC_HELPER_0(ObjFunc_0ARGS f)
+{
+    return 0;
+}
+static inline ObjFunc VALIDATE_FUNC_HELPER_1(ObjFunc_1ARGS f)
+{
+    return 0;
+}
+static inline ObjFunc VALIDATE_FUNC_HELPER_2(ObjFunc_2ARGS f)
+{
+    return 0;
+}
+static inline ObjFunc VALIDATE_FUNC_HELPER_3(ObjFunc_3ARGS f)
+{
+    return 0;
+}
+static inline ObjFunc VALIDATE_FUNC_HELPER_4(ObjFunc_4ARGS f)
+{
+    return 0;
+}
+static inline ObjFunc VALIDATE_FUNC_HELPER_5(ObjFunc_5ARGS f)
+{
+    return 0;
+}
+static inline ObjFunc VALIDATE_FUNC_HELPER_6(ObjFunc_6ARGS f)
+{
+    return 0;
+}
+
+#define VALIDATE_FUNC_0ARGS(func)                                            \
+    (0 ? VALIDATE_FUNC_HELPER_0(func) : (ObjFunc)func)
```

Comment:
Not wanting to be too picky, would this be simpler with a simple identity function, which passed it's argument back? I'd expect it to get optimised away as well.

Reply:
I tried this at first, but it doesn't work. Well, it *would* work in C++ code, and perhaps some C compilers would allow it as an extension, but in general, this results in a compiler error (in clang: `error: initializer element is not a compile-time constant`; in GCC: `error: initializer element is not constant`).

You can verify this with e.g. https://godbolt.org and this input, where the definition of `val` is accepted and that of `val2` produces a compiler error:
```C
static inline int func(int x) { return x; }

int val = 0 ? func(0) : 1;
int val2 = func(0);

int main(void)
{
    return val;
}
```
Perhaps I should explain that in a code comment? OTOH, if somebody tries to refactor the code, they'll quickly notice, I think...